views/my_packages.package.dt

```diff
@@ -19,6 +19,38 @@ block body
 				|= e.get!string
 				br
 
+	h2 General
+
+	.inputForm
+		form(method="POST", enctype="multipart/form-data", action="#{req.rootDir}my_packages/#{packageName}/set_logo")
+			h3 Logo
+			- string existing = pack["logo"].get!string;
+			- string logoPath = existing.length ? existing : (req.rootDir ~ "images/default-logo.png");
```

Comment:
hm I'm not sure if this is a good idea performance-wise. We certainly would need to cache it to avoid users from loading for minutes

Reply:
Yes, also for privacy reasons. It would just be fetched in the background as default icon, also means that content is OK from day one.
In particular this would prefer homepage over repository, so it can be customized, if people are doing that anyhow (e.g. for SEO/Social Media preview).
Just as motivation, this is what Firefox manages to extract from the browser history.
![og_image](https://user-images.githubusercontent.com/288976/35453292-ee173820-02c9-11e8-8f24-b26bd409dc92.png)
Seems like a better starting point that to give everyone the same logo.